src/internals/strings/impl/length.ts

```diff
@@ -0,0 +1,44 @@
+// Original impl https://gist.github.com/sno2/7dac868ec6d11abb75250ce5e2b36041
+
+type TCache = [string, 0[]];
+
+type LengthUp<
+  T extends string,
+  $Acc extends 0[] = [],
+  $Cache extends TCache[] = [[`$${string}`, [0]]]
+> =
+  //
+  $Cache extends [infer C extends TCache, ...infer $RestCache extends TCache[]]
+    ? (
+        `${C[0]}${C[0]}_` extends `$${string}$${infer $After}`
```

Comment:
The extra `$` is required as a hack to stop `${string}${string}` to be simplified to `${string}` as TypeScript will not simplify `string` parts together if they have a special character in the template string as well.